code/game/objects/items/shields.dm

```diff
@@ -27,32 +90,107 @@
+	var/can_strobe = TRUE //whenever or not you can convert those shields into flash shields
+	var/flashing = FALSE //if shield is flashing currently
 
-/obj/item/shield/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(transparent && (hitby.pass_flags & PASSGLASS))
-		return FALSE
-	if(attack_type == THROWN_PROJECTILE_ATTACK)
-		final_block_chance += 30
-	if(attack_type == LEAP_ATTACK)
-		final_block_chance = 100
+/obj/item/shield/riot/Initialize()
 	. = ..()
-	if(.)
-		on_shield_block(owner, hitby, attack_text, damage, attack_type)
+	if(flashmount_installed)
+		embedded_flash = new(src)
+		update_appearance()
+
+/obj/item/shield/riot/ComponentInitialize()
+	. = .. ()
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+/obj/item/shield/riot/attack(mob/living/M, mob/user)
+	if(user.istate.secondary)
+		return ..()
+	if(embedded_flash)
+		if(embedded_flash.burnt_out)
+			return ..()
+		embedded_flash.attack(M, user)
+		update_appearance(flash = TRUE)
+	else
+		. = ..()
 
 /obj/item/shield/riot/attackby(obj/item/W, mob/user, params)
 	if(istype(W, /obj/item/melee/baton))
 		if(cooldown < world.time - 25)
 			user.visible_message(span_warning("[user] bashes [src] with [W]!"))
 			playsound(user.loc, 'sound/effects/shieldbash.ogg', 50, TRUE)
 			cooldown = world.time
-	else if(istype(W, /obj/item/stack/sheet/mineral/titanium))
+		return
+
+	if(istype(W, fixing_material))
 		if (obj_integrity >= max_integrity)
 			to_chat(user, span_warning("[src] is already in perfect condition."))
 		else
-			var/obj/item/stack/sheet/mineral/titanium/T = W
+			var/obj/item/stack/sheet/T = W
 			T.use(1)
 			obj_integrity = max_integrity
 			to_chat(user, span_notice("You repair [src] with [T]."))
+		return
+
+	if(istype(W, /obj/item/wallframe/flasher) && can_strobe)
+		if(flashmount_installed)
+			to_chat("Flashbulb mount is already attached!")
+			return
+		to_chat(user, span_notice("You begin to attach the flashbulb mount..."))
+		if(do_after(user, 20, target = user))
+			if(!W || QDELETED(W))
+				return
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			flashmount_installed = TRUE
+			update_appearance()
+			qdel(W)
+		return
+
+	if(istype(W, /obj/item/assembly/flash/handheld) && flashmount_installed)
+		var/obj/item/assembly/flash/handheld/flash = W
+		if(flash?.burnt_out)
+			to_chat(user, span_warning("You feel like you want to reconsider putting a burnt out flashbulb inside!"))
+		to_chat(user, span_notice("You begin to replace the flashbulb..."))
+		if(do_after(user, 20, target = user))
+			if(!flash || QDELETED(flash))
+				return
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			if(embedded_flash)
+				embedded_flash.forceMove(get_turf(src))
+			embedded_flash = flash
+			flash.forceMove(src)
+			update_appearance()
+		return
+
+	if (W.tool_behaviour == TOOL_WIRECUTTER)
+		if(embedded_flash)
```

Comment:
you can make these into a single if statement

Reply:
done